exe/arduino_ci.rb

```diff
@@ -445,24 +460,6 @@ def perform_unit_tests(cpp_library, file_config)
   end
 end
 
-def build_shared_library(gcc_binary, platform, config, cpp_library)
-  attempt_multiline("Build shared library with #{gcc_binary} for #{platform}") do
-    exe = cpp_library.build_shared_library(
-      config.aux_libraries_for_unittest,
-      gcc_binary,
-      config.gcc_config(platform)
-    )
-    puts
-    unless exe
-      puts "Last command: #{cpp_library.last_cmd}"
-      puts cpp_library.last_out
-      puts cpp_library.last_err
-      return false
-    end
-    return true
```

Comment:
@jgfoster I think the actual problem was this `return true` which should just be an implicit return of `exe` 🤦   I'll push up this change in a bit and see what happens